devcycle_python_sdk/dvc_local_client.py

```diff
@@ -30,61 +30,54 @@ def __init__(self, sdk_key: str, options: DevCycleLocalOptions):
-                                                                                 self.local_bucketing)
-        self.event_queue_manager: EventQueueManager = EventQueueManager(sdk_key, self.options, self.local_bucketing)
-
-    def _add_platform_data_to_user(self, user: User) -> User:
-        user.platform = self._platform_data.platform
-        user.platformVersion = self._platform_data.platformVersion
-        user.sdkVersion = self._platform_data.sdkVersion
-        user.sdkType = self._platform_data.sdkType
-        return user
-
-    def _validate_sdk_key(self, sdk_key: str) -> None:
-        if sdk_key is None or len(sdk_key) == 0:
-            raise ValueError("Missing SDK key! Call build with a valid server SDK key")
-
-        if not sdk_key.startswith("server") and not sdk_key.startswith("dvc_server"):
-            raise ValueError(
-                "Invalid SDK key provided. Call build with a valid server SDK key"
-            )
-
-    def _validate_user(self, user: User) -> None:
-        if user is None:
-            raise ValueError("User cannot be None")
+        self.local_bucketing.set_platform_data(
+            json.dumps(self._platform_data.to_json())
+        )
 
-        if user.user_id is None or len(user.user_id) == 0:
-            raise ValueError("userId cannot be empty")
+        self.config_manager: EnvironmentConfigManager = EnvironmentConfigManager(
+            sdk_key, self.options, self.local_bucketing
+        )
+        self.event_queue_manager: EventQueueManager = EventQueueManager(
+            sdk_key, self.options, self.local_bucketing
+        )
 
-    def _is_initialized(self) -> bool:
-        return (self.config_manager and self.config_manager.is_initialized())
+    def is_initialized(self) -> bool:
+        return self.config_manager and self.config_manager.is_initialized()
 
     def set_client_custom_data(self, custom_data: Dict[str, Any]) -> None:
-        if not self._is_initialized():
+        """
+        Sets global custom data for this client. This data will be utilized in all segmentation and bucketing
+        decisions. This data will be merged with any custom data set on the user object, with user data
+        taking priority
+
+        :param custom_data: Global data to set.  Supported values are strings, numbers, booleans, and lists of strings,
+        numbers, and booleans. Nested dictionaries are not permitted
+        """
+        if not self.is_initialized():
             logger.debug("set_client_custom_data called before client has initialized")
             return
 
-        # TODO delegate to local bucketing api
-        pass
+        if custom_data:
+            try:
+                custom_data_json = json.dumps(custom_data)
+                self.local_bucketing.set_client_custom_data(custom_data_json)
+            except Exception as e:
+                logger.error("Error setting custom data: " + str(e))
```

Comment:
Do we want to handle this exception? Unless swallowing all exceptions from this method is what all the other SDKs do / the test harness expects?

Reply:
so it seems Go is the only SDK that actually throws up an error, everything else seems to do this consume and log pattern.  I'm personally on the fence for this kind of thing.  In go it makes sense as there is a standard error management pattern but in languages with exceptions it can be messier.

Reply:
Let's keep it as is then. I think we need to come up with a better way of safely surfacing errors from the local client and apply that to all the SDKs at some point.